pypdf/generic/_data_structures.py

```diff
@@ -270,6 +273,14 @@ def _clone(
                                     cur_obj.__class__(), pdf_dest, force_duplicate
                                 ),
                             )
+                            # check to see if we've previously processed our item
+                            if clon.indirect_reference is not None:
+                                idnum = clon.indirect_reference.idnum
+                                generation = clon.indirect_reference.generation
+                                if (idnum, generation) in visited:
+                                    cur_obj = None
+                                    break
```

Comment:
Should this continue rather than break?

Reply:
As you set `cur_obj` to `None` the while-loop would stop here anyway. You save one check and adding one element to the set. And it is clearer that from this point the loop is over.

I think it's good as-is :-) :+1: